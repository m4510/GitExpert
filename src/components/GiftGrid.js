import React from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifts } from '../hooks/useFetchGifts';

export const GiftGrid = ({ category }) => {
  const { loading, data: images } = useFetchGifts(category);

  return (
    <>
      <h3>{category}</h3>

      {loading && 'Loading...'}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
