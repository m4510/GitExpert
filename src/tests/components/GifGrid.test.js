import React from 'react';
import { shallow } from 'enzyme';
import { GiftGrid } from '../../components/GiftGrid';
import { useFetchGifts } from '../../hooks/useFetchGifts';
jest.mock('../../hooks/useFetchGifts');

describe('Test in Grid Component', () => {
  const category = 'Naruto';

  test('should render component correctly', () => {
    useFetchGifts.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GiftGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the component with the data', () => {
    const gifts = [
      {
        id: 1,
        title: 'Some title',
        url: 'https://localhost.com/images',
      },
      {
        id: 2,
        title: 'title 2',
        url: 'https://localhost.com/titles',
      },
    ];

    useFetchGifts.mockReturnValue({
      data: gifts,
      loading: false,
    });

    const wrapper = shallow(<GiftGrid category={category} />);

    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifts.length);
  });
});
