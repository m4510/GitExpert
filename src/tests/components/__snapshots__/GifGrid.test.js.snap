// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Grid Component should render component correctly 1`] = `
<Fragment>
  <h3>
    Naruto
  </h3>
  Loading...
  <div
    className="card-grid"
  />
</Fragment>
`;
